packages/shared/src/components/search/SearchBarSuggestion.tsx

```diff
@@ -7,11 +7,51 @@ import {
   ButtonSize,
 } from '../buttons/Button';
 import { AiIcon } from '../icons';
+import { AnalyticsEvent, Origin, TargetType } from '../../lib/analytics';
+import { SearchSession } from '../../graphql/search';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
 
-export const SearchBarSuggestion = <TagName extends AllowedTags>({
+export type SuggestionOrigin =
+  | Origin.HomePage
+  | Origin.HistoryTooltip
+  | Origin.HistoryPage;
+type SearchBarSuggestionProps = ButtonProps<AllowedTags> & {
+  suggestion?: Partial<SearchSession>;
+  origin?: SuggestionOrigin;
+};
+
+export const SearchBarSuggestion = ({
   className,
+  origin,
   ...props
-}: ButtonProps<TagName>): ReactElement => {
+}: SearchBarSuggestionProps): ReactElement => {
+  const { trackEvent } = useContext(AnalyticsContext);
+
+  useEffect(() => {
+    if (props.suggestion?.id) {
+      trackEvent({
+        event_name: AnalyticsEvent.Impression,
+        target_type: TargetType.SearchRecommendation,
+        target_id: props.suggestion.id,
+        feed_item_title: props.suggestion.prompt,
+        extra: JSON.stringify({ origin }),
+      });
+    }
+
+    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [props.suggestion?.id]);
```

Comment:
This should probably be empty to avoid it retriggering.
I'm also not sure if there is a case where it's not set.

Reply:
Ah is it to distinguish between history and recommendations?